buildpacktoml_test.go

```diff
@@ -0,0 +1,745 @@
+						Slices:      []layers.Slice{},
+					}); s != "" {
+						t.Fatalf("Unexpected:\n%s\n", s)
+					}
+				})
+
+				it("should include labels", func() {
+					mkfile(t,
+						"[[labels]]\n"+
+							`key = "some-key"`+"\n"+
+							`value = "some-value"`+"\n"+
+							"[[labels]]\n"+
+							`key = "some-other-key"`+"\n"+
+							`value = "some-other-value"`+"\n",
+						filepath.Join(appDir, "launch-A-v1.toml"),
+					)
+
+					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+					if err != nil {
+						t.Fatalf("Unexpected error:\n%s\n", err)
+					}
+
+					if s := cmp.Diff(br, lifecycle.BuildResult{
+						BOM: nil,
+						Labels: []lifecycle.Label{
+							{Key: "some-key", Value: "some-value"},
+							{Key: "some-other-key", Value: "some-other-value"},
+						},
+						MetRequires: nil,
+						Processes:   []launch.Process{},
+						Slices:      []layers.Slice{},
+					}); s != "" {
+						t.Fatalf("Unexpected:\n%s\n", s)
+					}
+				})
+
+				it("should include processes", func() {
+					mkfile(t,
+						`[[processes]]`+"\n"+
+							`type = "some-type"`+"\n"+
+							`command = "some-cmd"`+"\n"+
+							`[[processes]]`+"\n"+
+							`type = "other-type"`+"\n"+
+							`command = "other-cmd"`+"\n",
+						filepath.Join(appDir, "launch-A-v1.toml"),
+					)
+					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+					if err != nil {
+						t.Fatalf("Unexpected error:\n%s\n", err)
+					}
+					if s := cmp.Diff(br, lifecycle.BuildResult{
+						BOM:         nil,
+						Labels:      []lifecycle.Label{},
+						MetRequires: nil,
+						Processes: []launch.Process{
+							{Type: "some-type", Command: "some-cmd", BuildpackID: "A"},
+							{Type: "other-type", Command: "other-cmd", BuildpackID: "A"},
+						},
+						Slices: []layers.Slice{},
+					}); s != "" {
+						t.Fatalf("Unexpected metadata:\n%s\n", s)
+					}
+				})
+
+				it("should include slices", func() {
+					mkfile(t,
+						"[[slices]]\n"+
+							`paths = ["some-path", "some-other-path"]`+"\n",
+						filepath.Join(appDir, "launch-A-v1.toml"),
+					)
+
+					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+					if err != nil {
+						t.Fatalf("Unexpected error:\n%s\n", err)
+					}
+
+					if s := cmp.Diff(br, lifecycle.BuildResult{
+						BOM:         nil,
+						Labels:      []lifecycle.Label{},
+						MetRequires: nil,
+						Processes:   []launch.Process{},
+						Slices:      []layers.Slice{{Paths: []string{"some-path", "some-other-path"}}},
+					}); s != "" {
+						t.Fatalf("Unexpected:\n%s\n", s)
+					}
+				})
+			})
+		})
+
+		when("building succeeds with a clear env", func() {
+			it.Before(func() {
+				env.EXPECT().List().Return(append(os.Environ(), "TEST_ENV=cleared"))
+
+				bpTOML.Buildpack.Version = "v1.clear"
+				bpTOML.Path = filepath.Join(buildpacksDir, "A", "v1.clear")
+				bpTOML.Buildpack.ClearEnv = true // TODO: it is weird that all three of these conditions need to be specified
```

Comment:
It feels a bit weird that all three of these conditions need to be specified in order to get the test to behave correctly. I am not sure how to improve it...

Reply:
Yeah, I had that thought while reading. I think we could fix it with some refactoring of the test buildpacks but it's probably better to defer until we complete the package extraction refactor.